.github/workflows/go.yaml

```diff
@@ -108,31 +121,120 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
     - uses: actions/checkout@v2
-    - uses: docker/build-push-action@v2
       with:
-        context: .
-        file: ./Dockerfile
-        platforms: linux/amd64
-        push: false
+        path: ${{ env.git_repo_path }}
+        fetch-depth: 0  # also fetch tags
+    - name: Install podman
+      run: |
+        set -x
+        # Install nativelly when we have Ubuntu >= 20.10
+        . /etc/os-release
+        echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
+        curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/Release.key | sudo apt-key add -
+        sudo apt-get update
+        sudo apt-get -y install podman
+    - name: Build image
+      run: |
+        set -x
+        podman build --squash -f ./Dockerfile -t '${{ env.image_repo_ref }}:pr' .
+        podman images '${{ env.image_repo_ref }}:pr'
+        podman save '${{ env.image_repo_ref }}:pr' | lz4 - ~/operatorimage.tar.lz4
+    - name: Upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: operatorimage.tar.lz4
+        path: ~/operatorimage.tar.lz4
+        if-no-files-found: error
+        retention-days: ${{ env.retention_days }}
 
-  # TODO: Add e2e - use the same image sha from images step
+  test-e2e:
+    name: Test e2e
+    runs-on: ubuntu-20.04
+    needs: images
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: ${{ env.git_repo_path }}
+        fetch-depth: 0  # also fetch tags
+    - uses: actions/download-artifact@v2
+      with:
+        name: operatorimage.tar.lz4
+        path: ~/
+    - name: Load image
+      run: |
+        set -x
+         unlz4 ~/operatorimage.tar.lz4 - | docker load
+         # docker looses the registry part on save/load
+         docker tag "$( echo "${image_repo_ref}:pr" | sed -E -e 's~[^/]+/(.*)~\1~' )" "${image_repo_ref}:pr"
+         docker images '${{ env.image_repo_ref }}:pr'
+    - name: Setup go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.go_version }}
+    - uses: ./go/src/github.com/scylladb/scylla-operator/.github/actions/setup-minikube
```

Comment:
now that I've tried again (and it failed), I recalled that GH actions suck regarding var substitution and this is one of those contexts where it isn't supported :(

```
The workflow is not valid. .github/workflows/go.yaml (Line: 174, Col: 13): Unrecognized named-value: 'env'. Located at position 1 within expression: env.git_repo_path
```
https://github.com/scylladb/scylla-operator/actions/runs/636537964

Reply:
I meant to give step a string name, now CI displays:
![Selection_231](https://user-images.githubusercontent.com/3951334/110606162-320b3f80-818a-11eb-9b93-ea81d97b6c93.png)
where it could show "Setup minikube" or something like that